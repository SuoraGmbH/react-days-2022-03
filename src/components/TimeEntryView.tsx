import { TimeEntry } from "../domain/TimeEntry";

interface Props {
  timeEntry: TimeEntry;
}

const TimeEntryView = ({ timeEntry }: Props) => {
  return <div>{timeEntry.comment}</div>;
};

export default TimeEntryView;
