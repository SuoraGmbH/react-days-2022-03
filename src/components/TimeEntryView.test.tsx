import { render, screen } from "@testing-library/react";
import TimeEntryView from "./TimeEntryView";
import { TimeEntry } from "../domain/TimeEntry";

const timeEntry: TimeEntry = {
  id: "123213321",
  comment: "Hallo Jennifer",
  start: new Date("2022-03-23T15:30:30.526Z"),
  end: new Date("2022-03-23T15:45:30.526Z"),
};

describe("<TimeEntryView />", function () {
  test("renders without crashing", () => {
    render(<TimeEntryView timeEntry={timeEntry} />);
  });

  test("renders according to snapshot", () => {
    const { container } = render(<TimeEntryView timeEntry={timeEntry} />);
    expect(container).toMatchSnapshot();
  });

  test("renders the time entries comment", () => {
    render(<TimeEntryView timeEntry={timeEntry} />);
    screen.getByText("hallo Jennifer", { exact: false });
  });
});
