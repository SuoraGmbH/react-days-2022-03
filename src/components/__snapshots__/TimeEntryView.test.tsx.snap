// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeEntryView /> renders according to snapshot 1`] = `
<div>
  <div
    class="time-entry-view"
  >
    Hallo Jennifer
     (
    16:30:30
    -
    16:45:30
    )
  </div>
</div>
`;
